common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -554,6 +568,49 @@ def should_force_account_creation():
         # We know they did not just login, because the login process rejects unverified users.
 
 
+def create_user(strategy, details, user=None, autoprovision=False, *args, **kwargs):
+    """
+    Creates user passing extra fields if in autoprovisioning mode
+    """
+    if user:
+        return {'is_new': False}
+
+    fields = dict(
+        (name, kwargs.get(name) or details.get(name))
+        for name in (strategy.setting('USER_FIELDS', USER_FIELDS))
+    )
+
+    if not fields:
+        return
+
+    if not autoprovision:
+        return {
+            'is_new': True,
+            'user': strategy.create_user(**fields)
+        }
+
+    extra_fields = dict(
+        (name, kwargs.get(name) or details.get(name, None))
+        for name in (strategy.setting('EXTRA_USER_FIELDS', []))
+    )
+
+    fields.update(extra_fields)
+    fields['autoprovision'] = True
+
+    try:
+        # Mocking partial pipeline request so that create_account_with_params used in strategy to create a user
+        # correctly get backend and provider settings
+        save_status_to_session(strategy, details=details, user=user, autoprovision=autoprovision, *args, **kwargs)
```

Comment:
This is a little bit ugly/confusing, and the only reason it's needed is to make `create_account_with_params` set a password automatically.

Suggestion: Remove `save_status_to_session` and the `try`/`finally`. Then in `strategy.create_user` just add the line `user_fields['password'] = pipeline.make_random_password()` before you call `create_account_with_params()`. It should work just as well but is easier to understand and doesn't require us to fake a partial pipeline when the pipeline is really still running. It's also fewer lines of changed code.